enclave/core/sgx/switchless_t_wrapper.c

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#define OE_NEED_STDC_NAMES
+
+#include <openenclave/enclave.h>
+
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/edger8r/enclave.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/thread.h>
+
+/* Rename the global ecalls table. */
+#define __oe_ecalls_table __oe_switchless_ecalls_table
+#define __oe_ecalls_table_size __oe_switchless_ecalls_table_size
+
+/* Override oe_call_host_function() calls with _call_host_function(). */
+#define oe_call_host_function _call_host_function
+
+/* Use this function below instead of oe_call_host_function(). */
+static oe_result_t _call_host_function(
+    size_t function_id,
+    const void* input_buffer,
+    size_t input_buffer_size,
+    void* output_buffer,
+    size_t output_buffer_size,
+    size_t* output_bytes_written)
+{
+    return oe_call_host_function_by_table_id(
+        OE_SWITCHLESS_OCALL_FUNCTION_TABLE_ID,
+        function_id,
+        input_buffer,
+        input_buffer_size,
+        output_buffer,
+        output_buffer_size,
+        output_bytes_written,
+        false /* non-switchless */);
+}
+
+#include "switchless_t.c"
```

Comment:
Including a .c file is very unusual. 
I'd recommend adding a /* comment */ explaining why you're doing it here.

Reply:
Done. Added a comment indicating that the file is being included so that the generated code is customized for internal use via the macros defined above.